### DISCUSSION TITLE
What Should I Name My Bevy Plugin?

### DISCUSSION THREAD

**Discussion by zicklag** [bevyengine/bevy#1202] [Q&A] [answered]
Hey there, I am wanting to make a bevy plugin to integrate [LDtk](https://github.com/deepnight/ldtk) maps, but I wanted to double-check whether or not it was OK to name my crate `bevy_ldtk` or if I should name it something else because it isn't an official bevy crate.

I know there are crates out there already like `bevy_tiled` that aren't official, but I thought I could remember somebody mentioning something like making `bevy_prototype_something` crates and I wasn't sure if that ever turned into a thing or not.

Is there something specific I should name my bevy plugin?

**Comment by cart** [comment, marked as answer]
Haha yeah there was a long conversation about naming conventions.

In short: I think people should use `bevy_xxx`, with the caveat "please be reasonable". If you are about to claim a generic name like `bevy_animation`, `bevy_color`, or `bevy_editor` ... please ask first. 

1. I want people to have "nice" names for their projects. bevy_contrib_xxx is probably what I would pick as an alternative, but even that makes me "unhappy". 
2. We already live in a world where people can take whatever crate name they want.  Lets say we have bevy_contrib_xxx as the approved format. People can still take bevy_xxx. Why should the good actors have worse names than the bad actors? Seems pretty unfair to me. 
3. We have already claimed most of the names we will need in the bevy_xxx namespace. We can always claim more if we need to. And if someone ever claims one we need, we can always try to work something out with the crate owner.
4. If we stake a claim to all bevy_xxx crates, then in general people will assume that bevy_xxx is "official". This could create problems were people assume 3rd party crates are "official" when they actually arent

**Reply by zicklag** [reply]
OK, great, thanks!

Somewhat related, we might get crates as namespaces sometime soon, which might do a lot to help this one day:

https://internals.rust-lang.org/t/pre-rfc-packages-as-optional-namespaces/13059/71?u=zicklag

Then the official ones become `bevy/renderer`, but for now, I think your on the right track.

**Reply by cart** [reply]
Ooh that makes me so happy. I hope it gets merged.

**Reply by Restioson** [reply]
By the way, this can be tried out (not on official crates.io though) using the instructions [here](https://github.com/Manishearth/namespacing-rfc/issues/10)

**Comment by mockersf** [comment]
Not about names, but about guidelines for plugin authors: could we agree on metadata about plugins?
First use would be to mention bevy version supported, and plugin version (keeping track of previous versions of bevy for people who don't want to update), something like
|bevy|my-plugin|
|---|---|
|0.3|0.1|
|0.4|0.3|

Second would be relationship to master:
- plugin author will track master in their development branch
- plugin author welcome PR to keep up to date with master
- plugin author don't want to track master
- ...

Next could be informations about systems added by the plugin, components, stages, ...

**Reply by zicklag** [reply]
That's a great idea. If we created a README template, that would be really helpful to plugin authors and would make it more likely that people actually add the information.

We could create template badges for the master relationship, etc., too.

**Reply by cart** [reply]
Yeah those are good ideas :)
